src/lib.rs

```diff
@@ -139,45 +139,16 @@ where
 }
 
 impl Client<ClientManager> {
-    fn steam_api_init_ex(p_out_err_msg: *mut SteamErrMsg) -> ESteamAPIInitResult {
-        let versions: Vec<&[u8]> = vec![
-            sys::STEAMUTILS_INTERFACE_VERSION,
-            sys::STEAMNETWORKINGUTILS_INTERFACE_VERSION,
-            sys::STEAMAPPS_INTERFACE_VERSION,
-            sys::STEAMCONTROLLER_INTERFACE_VERSION,
-            sys::STEAMFRIENDS_INTERFACE_VERSION,
-            sys::STEAMGAMESEARCH_INTERFACE_VERSION,
-            sys::STEAMHTMLSURFACE_INTERFACE_VERSION,
-            sys::STEAMHTTP_INTERFACE_VERSION,
-            sys::STEAMINPUT_INTERFACE_VERSION,
-            sys::STEAMINVENTORY_INTERFACE_VERSION,
-            sys::STEAMMATCHMAKINGSERVERS_INTERFACE_VERSION,
-            sys::STEAMMATCHMAKING_INTERFACE_VERSION,
-            sys::STEAMMUSICREMOTE_INTERFACE_VERSION,
-            sys::STEAMMUSIC_INTERFACE_VERSION,
-            sys::STEAMNETWORKINGMESSAGES_INTERFACE_VERSION,
-            sys::STEAMNETWORKINGSOCKETS_INTERFACE_VERSION,
-            sys::STEAMNETWORKING_INTERFACE_VERSION,
-            sys::STEAMPARENTALSETTINGS_INTERFACE_VERSION,
-            sys::STEAMPARTIES_INTERFACE_VERSION,
-            sys::STEAMREMOTEPLAY_INTERFACE_VERSION,
-            sys::STEAMREMOTESTORAGE_INTERFACE_VERSION,
-            sys::STEAMSCREENSHOTS_INTERFACE_VERSION,
-            sys::STEAMUGC_INTERFACE_VERSION,
-            sys::STEAMUSERSTATS_INTERFACE_VERSION,
-            sys::STEAMUSER_INTERFACE_VERSION,
-            sys::STEAMVIDEO_INTERFACE_VERSION,
-            b"\0",
-        ];
-
-        let merged_versions: Vec<u8> = versions.into_iter().flatten().cloned().collect();
-        let merged_versions_ptr = merged_versions.as_ptr() as *const ::std::os::raw::c_char;
-
-        unsafe { sys::SteamInternal_SteamAPI_Init(merged_versions_ptr, p_out_err_msg) }
-    }
-
-    /// Attempts to initialize the steamworks api and returns
-    /// a client to access the rest of the api.
+    /// Call to the native SteamAPI_Init function.
+    /// should not be used directly, but through either
+    /// init_flat() or init_flat_app()
+    fn steam_api_init_flat(p_out_err_msg: *mut SteamErrMsg) -> ESteamAPIInitResult {
```

Comment:
Since this function writes to the raw pointer, it should be marked `unsafe`, or preferably take `&mut SteamErrMsg` to ensure safety.

Reply:
Hello,

How do you want me to mark the function as unsafe? do we add crate unsafe_fn?

Function was written the same way the steam_api_init_ex one was, just without the required shenanigans to build the extra parameter required for SteamInternal_SteamAPI_Init.

Reply:
Either mark the function with the `unsafe` keyword, like
```rs
unsafe fn steam_api_init_flat(...
```
or make it take a mutable reference:
```rs
fn steam_api_init_flat(p_out_err_msg: &mut SteamErrMsg) ...
```
That will make it safe to use, because it'll statically require you to have proper access to the error buffer. As of right now, you could call this function with `std::ptr::null_mut()` as the parameter for example, without any unsafe at the callsite. Keep in mind that constructing invalid _raw pointers_ is perfectly legal in safe Rust. Only when they are accessed through `std::ptr::read/write`, do they need to be valid (point to proper allocation / not be null / be aligned)

Reply:
Hello,

I flagged the function as unsafe with commit fc37868 :)